### PR TITLE
GH-34988: [C#] Fix Windows-specific test issue in CDataSchemaPythonTest

### LINKED ISSUE #34988: Unable to run C# CDataSchemaPythonTests under Windows
### Describe the bug, including details regarding any error messages, version, and platform.

The C# CDataSchemaPythonTests don't run under Windows due to a quirk in the way embedded Python initializes sys.path under Windows.

### Component(s)

C#

### PR DESCRIPTION
**Pull request by CurtHagenlocher**
### Rationale for this change

Fixes issue described in referenced bug.

### What changes are included in this PR?

Changes test initialization under Windows to add a references to the "DLLs" subdirectory containing _socket.pyd.

### Are these changes tested?

Covered by existing tests.

### Are there any user-facing changes?

No.
* Closes: #34988

### CODE REVIEW

File: csharp/src/Apache.Arrow/C/CArrowSchemaExporter.cs [hunk 1]
```diff
@@ -86,8 +88,7 @@ public static unsafe void ExportField(Field field, CArrowSchema* schema)
         {
             ExportType(field.DataType, schema);
             schema->name = StringUtil.ToCStringUtf8(field.Name);
-            // TODO: field metadata
-            schema->metadata = null;
+            schema->metadata = field.HasMetadata ? ExportMetadata(field.Metadata) : null;
```
Comment (CurtHagenlocher) [inline csharp/src/Apache.Arrow/C/CArrowSchemaExporter.cs:91]: oops; hadn't meant to include these changes yet.

**Comment by CurtHagenlocher** [comment]
It turned out that both my local editor and the browser-hosted GitHub editor really wanted that trailing newline in the file I was trying to revert... .

A longer description of the need for the change can be found at https://github.com/pythonnet/pythonnet/issues/2143.

**Comment by wjones127** [comment]
Thanks for submitting this @CurtHagenlocher. Could you remind me why we don't have these tests skipped or failed in Windows CI?

**Comment by CurtHagenlocher** [comment]
> Thanks for submitting this @CurtHagenlocher. Could you remind me why we don't have these tests skipped or failed in Windows CI?

I have to admit that I can't tell based on a cursory examination of the CI scripts how this could be working there. Is it the expectation that testing locally on dev boxes would require the use of the CI scripts?

**Comment by wjones127** [comment]
> I have to admit that I can't tell based on a cursory examination of the CI scripts how this could be working there. Is it the expectation that testing locally on dev boxes would require the use of the CI scripts?

No, it's just that, whenever possible, I'd like for our CI to be capable of catching regressions in any fixes we make. Otherwise, can't always guarantee we won't break this again.

I suppose something in find-libpython might be handling this, or else there's something in the GitHub runner's Python environment keeping it happy.

**Review by wjones127 — APPROVED** [review]

**Pull request merged**